rag/sql_rag/r2r_vector_store.py

```
@@ -0,0 +1,318 @@
+            response.raise_for_status()
+            # Handle cases where response might be empty
+            if response.status_code == 204 or not response.content:
+                return {}
+            return response.json()
+        except requests.exceptions.HTTPError as e:
+            # Handle 409 Conflict for duplicate documents gracefully
+            if response.status_code == 409 and method == "POST" and endpoint == "documents":
+                print(f"Document already exists (409): {kwargs.get('files', '')}")
+                return {"status": "exists"}
+            print(f"Error: {e}")
+            raise Exception(f"Failed to run request: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)